### PR TITLE
Nested aggregations

### PR DESCRIPTION
**Pull request by berggren**
This PR changes how we do aggregations. It uses nested aggregations instead of computing all buckets server side, making it faster and less memory hungry.

Also some dafe guards in place to not render and calculate charts when there are many (>300k) events in the result.

### CODE REVIEW

**Comment by berggren** [comment]
Fixes #446

**Comment by aarontp** [comment]
Sorry for the delay, I'll take a look at this today.

File: timesketch/lib/aggregators.py [hunk 1]
```diff
@@ -13,6 +13,9 @@
 # limitations under the License.
 """Elasticsearch aggregations."""
 
+# Max number of search hits to chart
+MAX_RESULT_LIMIT = 300000
```
Comment (aarontp) [inline timesketch/lib/aggregators.py]: Just curious; How did you come up with this limit?

File: timesketch/ui/explore/filter.directive.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ export const tsFilter = function () {
                 scope.meta.es_total_count = 0
                 scope.meta.es_time = 0
                 scope.meta.noisy = false
+                ctrl.search(scope.query, scope.filter, scope.queryDsl)
```
Comment (aarontp) [inline timesketch/ui/explore/filter.directive.ts]: I'll have to ask you more about this later.  I don't really know what this does.

File: timesketch/lib/aggregators.py (hunk 1 above)
Comment (berggren) [reply timesketch/lib/aggregators.py]: Good question, I did some simple memory monitoring to see the sweetspot for most setups. This should be moved to config though. Will add todo.

File: timesketch/lib/aggregators.py [hunk 3]
```diff
@@ -29,12 +32,52 @@ def heatmap(es_client, sketch_id, query_string, query_filter, query_dsl,
...
     aggregation = {
-        u'heatmap': {
-            u'date_histogram': {
-                u'field': u'datetime',
-                u'interval': u'hour',
-                u'format': u'e,H'
+        u'byDay': {
+            u'terms': {
+                u'script': {
+                    u'file': 'dateaggregation',
```
[the remark this answers is not in the document]
Comment (berggren) [reply timesketch/lib/aggregators.py]: Correct.

File: timesketch/ui/explore/filter.directive.ts (hunk 2 above)
Comment (berggren) [reply timesketch/ui/explore/filter.directive.ts]: This is how AngularJS 1.x works. Happy to explain more later.

**Pull request merged**